pkg/syncer/spec/mutators/deployment.go

```diff
@@ -207,6 +211,25 @@ func (dm *DeploymentMutator) Mutate(obj *unstructured.Unstructured) error {
 		templateSpec.Volumes = append(templateSpec.Volumes, serviceAccountVolume)
 	}
 
+	if dm.syncTargetLogicalClusterName == upstreamLogicalName {
```

Comment:
When would this be true? False? Why only do this when the deployment's logical cluster == the logical cluster where the SyncTarget lives?

Reply:
That's because this PR provides only a limited implementation, and does not support the upcoming use-case where user workspaces and location workspaces are distinct.

See the following comment: https://github.com/kcp-dev/kcp/pull/1708#issuecomment-1243915910 and the comments before that led to this decision. The followup issue to support the more general case is here: https://github.com/kcp-dev/kcp/issues/1987

Reply:
Could we please add a comment to the code itself with this information? Otherwise, it's not very clear.

Reply:
@lionelvillard Could you add a TODO here ?

Reply:
@davidfestal done!